zypp/target/RpmPostTransCollector.cc

```diff
@@ -52,7 +52,7 @@ namespace zypp
 	{ if ( !_scripts.empty() ) discardScripts(); }
 
 	/** Extract and remember a packages %posttrans script for later execution. */
-	bool collectScriptFromPackage( ManagedFile rpmPackage_r )
+        bool collectScriptFromPackage( ManagedFile rpmPackage_r, const std::string &scriptArg )
```

Comment:
Why do we still need `scriptArg` when collecting the script inside the transaction?

According to the pastebin I posted on IRC, installing 3 versions of a package, all 3 %posttrans get a 3 as argument. So it's the number of instances installed when executing the script, not when collecting it.

Reply:
Because when the scripts are executed all we know is the path to the script. 
But I guess we could just save the Pooltem::name together with the script?

Reply:
Yes. Can be done _in_ `collectScriptFromPackage`. It rerieves the `RpmHeader` anyway, so you can get and save the pkg name.

Reply:
Done